back-end/src/main/java/server/FirstRequestHandler.java

```diff
@@ -19,7 +20,10 @@
     private final Gson gson;
 
     public FirstRequestHandler() {
-        gson = new GsonBuilder().create();
+        gson =
+                new GsonBuilder()
+                        .addSerializationExclusionStrategy(new AnnotationExclusionStrategy())
+                        .create();
     }
```

Comment:
`gson.create()` exists in all requestHandler files. Most of them do not need special exclusion, but do you think `gson.fromJSON` in `ProjectRequestHandler` need exclusion?

Reply:
I don't think `@Exclude` is needed in deserialization. I think `@Exclude` is most important to any request handler that serialize the entire project. Also I only set the `@Exclude` to be a serialization strategy, not a deserialization one.
ProjectRequestHandler does require `RuntimeTypeAdapterFactory` though. It will be in the PR of hierarchy.